packages/vincent-rest-api/src/tests/integration.test.ts

```diff
@@ -0,0 +1,255 @@
+import { Api, ApiError } from '../common/api/index';
+import { ICreateAppDef } from '../common/api/models/ICreateAppDef';
+import { VersionChanges } from '../common/api/models/VersionChanges';
+import { ICreateAppVersionDef } from '../common/api/models/ICreateAppVersionDef';
+
+const fail = (message: string) => {
+  expect(message).toBeFalsy();
+};
+
+type ValidationErrors = Array<{
+  path: string[];
+  message: string;
+}>;
+
+const waitBetweenRequests = () => new Promise((r) => setTimeout(r, 100));
+
+describe('Integration tests', () => {
+  const apiClient = new Api({
+    BASE: 'http://localhost:3000',
+  });
+
+  let appIdentity: string;
+  let versionIdentity: string;
+
+  // App endpoint tests
+  describe('App endpoints', () => {
+    test('create app - valid', async () => {
+      const app: ICreateAppDef = {
+        name: 'Integration Test App',
+        description: 'An app for integration testing',
+        contactEmail: 'test@example.com',
+        appUserUrl: 'https://example.com',
+        logo: 'base64encodedlogo',
+        redirectUris: ['https://example.com/callback'],
+        deploymentStatus: ICreateAppDef.deploymentStatus.DEV,
+        managerAddress: '0xa723407AdB396a55aCd843D276daEa0d787F8db5',
+      };
+
+      const response = await apiClient.app.createApp(app);
+
+      if ('appId' in response) {
+        expect(response).toHaveProperty('identity');
+        expect(response).toHaveProperty('appId');
+        expect(response.name).toBe(app.name);
+        expect(response.description).toBe(app.description);
+
+        appIdentity = response.identity;
+        console.log('Created app with identity:', appIdentity);
+      } else {
+        fail('Expected AppDef response but got Error');
+      }
+
+      await waitBetweenRequests();
+    });
+
+    test('create app - invalid schema', async () => {
+      const invalidApp = {
+        name: 'Invalid Test App',
+      };
+
+      try {
+        await apiClient.app.createApp(invalidApp as any);
+        fail('Expected API call to fail but it succeeded');
+      } catch (e: any) {
+        if (e instanceof ApiError) {
```

Comment:
This is a lot of boilerplate -- don't test for errors like this, it's very brittle, hard to reason about, and verbose. What you want is something like

```
const result = await expect(apiClient.app....).rejects.toThrow();
```